underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -183,7 +188,15 @@ public FileSystem load(String userKey) throws Exception {
           // Set the class loader to ensure FileSystem implementations are
           // loaded by the same class loader to avoid ServerConfigurationError
           Thread.currentThread().setContextClassLoader(currentClassLoader);
-          return path.getFileSystem(hdfsConf);
+
+          if (UserGroupInformation.isSecurityEnabled() && !userKey.equals(HDFS_USER)) {
```

Comment:
should we add a property to enable/disable the ufs impersonation?   I'm just thinking some of our users are just using alluxio user as a super user to read/write everything,  so if we enable the impersonation as default, it might break their current data access model.

Reply:
+1, I have the same concern

Reply:
make sense